### PR TITLE
Interface `icx_getorder` RPC

### PR DESCRIPTION
**Pull request by surangap**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
This PR will interface `icx_getorder` RPC to jellyfish

#### Which issue(s) does this PR fixes?:

Fixes #

#### Additional comments?:
depends on #400

### CODE REVIEW

File: packages/jellyfish-api-core/__tests__/category/icxorderbook/getOrder.test.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    }
+    const createOrderResult: ICXGenericResult = await client.icxorderbook.createOrder(order, [])
+    const createOrderTxId = createOrderResult.txid
+    await container.generate(1)
+
+    // get order createOrderTxId and check
+    const retrievedOrder: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.icxorderbook.getOrder(createOrderTxId)
+    expect((retrievedOrder as Record<string, ICXOrderInfo>)[createOrderTxId]).toStrictEqual(
+      {
+        // status: ICXOrderStatus.OPEN, //NOTE(surangap): status is not returned?
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/icxorderbook/getOrder.test.ts:52]: will be fixed with https://github.com/DeFiCh/ain/pull/571
will uncomment then

File: packages/jellyfish-api-core/__tests__/category/icxorderbook/getOrder.test.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    const makeOffer2Result = await client.icxorderbook.makeOffer(offer2, [])
+    const makeOffer2TxId = makeOffer2Result.txid
+    await container.generate(1)
+
+    // retrive makeOffer2TxId
+    const retrievedOrder: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.icxorderbook.getOrder(makeOffer2TxId)
+    expect((retrievedOrder as Record<string, ICXOfferInfo>)[makeOffer2TxId]).toStrictEqual(
+      {
+        orderTx: createOrder2TxId,
+        status: ICXOrderStatus.EXPIRED, // NOTE(surangap): why this is EXPIRED ? should be OPEN?
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/icxorderbook/getOrder.test.ts:151]: will be fixed with https://github.com/DeFiCh/ain/pull/571
will uncomment then

**Review by canonbrother — APPROVED** [review]

File: website/docs/jellyfish/api/icxorderbook.md [hunk 3]
```diff
@@ -92,3 +92,43 @@ interface ICXGenericResult {
...
 }
 ```
+
+## getOrder
+
+Returns information about order or fillorder
+
+```ts title="client.icxorderbook.getOrder()"
+interface icxorderbook {
+  getOrder (orderTx: string): Promise<Record<string, ICXOrderInfo | ICXOfferInfo>>
```
Comment (fuxingloh) [inline website/docs/jellyfish/api/icxorderbook.md:102]: `ICXOrderInfo | ICXOfferInfo` is such a mind fuck, I always thought they were the same interface without realizing one is "Order", another is "Offer"

**Review by fuxingloh — APPROVED** [review]

File: website/docs/jellyfish/api/icxorderbook.md (hunk 3 above)
Comment (surangap) [reply website/docs/jellyfish/api/icxorderbook.md:102]: yeah. the same function `getOrder()`returning both order and offer having different structures. not the best way to do it. gave some trouble when interfacing also.

**Pull request merged**